va/va.go

```diff
@@ -173,13 +173,20 @@ func (va ValidationAuthorityImpl) getAddr(ctx context.Context, hostname string)
 	return addr, addrs, nil
 }
 
-type dialer struct {
-	record core.ValidationRecord
-	stats  metrics.Scope
+type http01Dialer struct {
+	record      core.ValidationRecord
+	stats       metrics.Scope
+	dialerCount int
 }
 
-func (d *dialer) Dial(_, _ string) (net.Conn, error) {
-	realDialer := net.Dialer{Timeout: validationTimeout}
+func (d *http01Dialer) newDialer() net.Dialer {
+	// Record that we created a new dialer
+	d.dialerCount++
+	return net.Dialer{Timeout: validationTimeout}
+}
+
+func (d *http01Dialer) Dial(_, _ string) (net.Conn, error) {
+	realDialer := d.newDialer()
```

Comment:
I think this would be clearer if you moved it to right above `realDialer.Dial(...)`, even though that means duplicating it in the two different `if` blocks.